### PR TITLE
Encoder support for TimestampedLwM2mNodes

### PR DESCRIPTION
**Pull request by adamsero**

### CODE REVIEW

**Comment by sbernard31** [comment]
To contribute to Eclipse foundation project, you need to sign an Eclipse Contributor Agreement using the email used in your commit.

I just check for adam*****ski@****.com and it seems there is no valid ECA.

See [How to contribute code§Legal stuff](https://github.com/eclipse/leshan/wiki/How-to-contribute#legal-stuff-) for more details. 

Do not hesitate to ask if you have any issue.

**Comment by adamsero** [comment]
The solution for #1228

**Comment by adamsero** [comment]
I used Eclipse to format, should be fine now.

File: leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/senml/LwM2mNodeSenMLEncoder.java [hunk 1]
```diff
@@ -30,11 +30,13 @@
...
 import org.eclipse.leshan.core.node.LwM2mResourceInstance;
 import org.eclipse.leshan.core.node.ObjectLink;
+import org.eclipse.leshan.core.node.TimestampedLwM2mNodes;
 import org.eclipse.leshan.core.node.TimestampedLwM2mNode;
 import org.eclipse.leshan.core.node.codec.CodecException;
 import org.eclipse.leshan.core.node.codec.LwM2mValueConverter;
 import org.eclipse.leshan.core.node.codec.MultiNodeEncoder;
 import org.eclipse.leshan.core.node.codec.TimestampedNodeEncoder;
+import org.eclipse.leshan.core.node.codec.TimestampedMultiNodeEncoder;
 import org.eclipse.leshan.core.util.Validate;
```
Comment (sbernard31) [inline leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/senml/LwM2mNodeSenMLEncoder.java:40]: There is some bad order organize import.
You can use eclipse to organize import.

**Comment by sbernard31** [comment]
I fixed some cosmetics issue : 
- a new formatting issue in DefaultLwM2mEncoder
- javadoc in TimestampedMultiNodeEncoder (DefaultLwM2mDecoder => DefaultLwM2mEncoder)

I also fixed something which looks like a bug introduced with your last commit : 
```diff
if (node != null) {
    node.accept(internalEncoder);

-    SenMLRecord record = internalEncoder.records.get(0);
-    record.setBaseTime(timestamp);
-    pack.addRecord(record);

+    List<SenMLRecord> records = internalEncoder.records;
+    if (!records.isEmpty()) {
+        records.get(0).setBaseTime(timestamp);
+        pack.addRecords(records);
+    }
}
```

I also adapted the test case to detect this issue :point_up:  :
```java
 @Test
public void senml_json_encode_timestamped_nodes() throws CodecException {
    TimestampedLwM2mNodes timestampedLwM2mNodes = TimestampedLwM2mNodes.builder()
            .put(500_000_001L, new LwM2mPath(0, 0, 0), LwM2mSingleResource.newStringResource(0, "TestString"))
            .put(500_000_002L, new LwM2mPath(0, 1),
                    new LwM2mObjectInstance(1, Arrays.asList(LwM2mSingleResource.newBooleanResource(1, true),
                            LwM2mSingleResource.newIntegerResource(2, 123))))
            .build();

    byte[] encoded = encoder.encodeTimestampedNodes(timestampedLwM2mNodes, ContentFormat.SENML_JSON, model);

    String expectedString = new StringBuilder()
            .append("[{\"bn\":\"/0/0/0\",\"bt\":500000001,\"vs\":\"TestString\"},") //
            .append("{\"bn\":\"/0/1/\",\"bt\":500000002,\"n\":\"1\",\"vb\":true},") //
            .append("{\"n\":\"2\",\"v\":123}]") //
            .toString();

    Assert.assertEquals(expectedString, new String(encoded));
}
```

**Comment by sbernard31** [comment]
This is now integrated in `master` (commit a759c24ebc1a01017e54895a3f414a94784257f9)

**Pull request closed without merging**